jrnl/args.py

```diff
@@ -17,6 +17,48 @@
 from .plugins import util
 
 
+def deserialize_config_args(input: list) -> dict:
+
+    """
+
+    Convert a two-element list of configuration key-value pair into a flat dict
+
+    :param input: list of configuration keys in dot-notation and their respective values.
+    :type input: list
+    :return: A single level dict of the configuration keys in dot-notation and their respective desired values
+    :rtype: dict
+    """
+
+    assert len(input) == 2
+    runtime_modifications = {}
+
+    cfg_key = input[0]
+    cfg_value = input[1]
+    cfg_value = cfg_value.strip()
+
+    # Convert numbers and booleans
```

Comment:
Instead of doing our own type coercion here, we should rely on the yaml library we already use. What if we don't choose a type until we know where the value is going? Then we can match the type to the destination (which has already been parsed by the yaml library).

Reply:
I'm forwarding the input to `yaml.load` and returning that output. Let me know if that is not the direction you'd like to take it and we can have a discussion on implementing it